DeliveryApi/cdk/src/main/java/com/ilmlf/delivery/db/DbStack.java

```diff
@@ -0,0 +1,249 @@
+import software.amazon.awscdk.services.ec2.SecurityGroup;
+import software.amazon.awscdk.services.ec2.SecurityGroupProps;
+import software.amazon.awscdk.services.ec2.SubnetSelection;
+import software.amazon.awscdk.services.ec2.SubnetType;
+import software.amazon.awscdk.services.ec2.Vpc;
+import software.amazon.awscdk.services.iam.Effect;
+import software.amazon.awscdk.services.iam.PolicyStatement;
+import software.amazon.awscdk.services.iam.PolicyStatementProps;
+import software.amazon.awscdk.services.iam.Role;
+import software.amazon.awscdk.services.iam.RoleProps;
+import software.amazon.awscdk.services.iam.ServicePrincipal;
+import software.amazon.awscdk.services.rds.Credentials;
+import software.amazon.awscdk.services.rds.DatabaseInstance;
+import software.amazon.awscdk.services.rds.DatabaseInstanceEngine;
+import software.amazon.awscdk.services.rds.DatabaseInstanceProps;
+import software.amazon.awscdk.services.rds.DatabaseProxy;
+import software.amazon.awscdk.services.rds.DatabaseProxyProps;
+import software.amazon.awscdk.services.rds.MySqlInstanceEngineProps;
+import software.amazon.awscdk.services.rds.MysqlEngineVersion;
+import software.amazon.awscdk.services.rds.ProxyTarget;
+import software.amazon.awscdk.services.rds.StorageType;
+import software.amazon.awscdk.services.secretsmanager.ISecret;
+import software.amazon.awscdk.services.secretsmanager.Secret;
+import software.amazon.awscdk.services.secretsmanager.SecretProps;
+import software.amazon.awscdk.services.secretsmanager.SecretStringGenerator;
+
+/**
+ * Database stack contains the followings.
+ * 1. Network resources (VPC, subnets, and Security Group)
+ * 2. RDS Database
+ * 3. RDS Proxy for connecting to database
+ * 4. Secrets for Lambda to connect to RDS Proxy
+ */
+@Getter
+public class DbStack extends Stack {
+  /**
+   * Database username for DB Proxy connection (via IAM Authorization).
+   */
+  private final String user;
+  private final String instanceEndpoint;
+  private final String proxyEndpoint;
+  private final Vpc vpc;
+  private final SecurityGroup securityGroup;
+  private final String proxyArn;
+
+  /**
+   * Contains database info, admin username, and password. This is a secret generated when
+   * creating the DB. Example of keys are:
+   * - username (admin username)
+   * - password (admin password)
+   * - engine (mysql
+   * - port (3306)
+   * - dbname (FarmerDB)
+   * - host (used for connection)
+   */
+  private final ISecret adminSecret;
+
+  /**
+   * Secret to beused by Lambda to connect to RDS Proxy via IAM authentication.
+   * - username (hardcoded to lambda_iam)
+   * - password
+   */
+  private final Secret userSecret;
+
+  /**
+   * Create a Database stack.
+   *
+   * @param scope used by superclass.
+   * @param id used by superclass.
+   * @param props used by superclass.
+   */
+  public DbStack(final Construct scope, final String id, final StackProps props) {
+    super(scope, id, props);
+
+    this.user = "lambda_iam";
```

Comment:
We should avoid hardcoded username and put it in a context variable

Reply:
Context variable "dbUsername" added, if none found will resort to default "lambda_iam"

Reply:
Will change. The user name will be read from `cdk.json` . If not provided, it will default to this value.